src/json.rs

```diff
@@ -98,13 +97,115 @@ fn can_get_selected_value() {
     assert_eq!(result, Some(&Value::Bool(true)));
 }
 
-/// Create [`TreeItem`]s from a [JSON](Value).
-#[must_use]
-pub fn tree_items(root: &Value) -> Vec<TreeItem<'_, Selector>> {
-    match root {
-        Value::Object(object) => from_object(object),
-        Value::Array(array) => from_array(array),
-        _ => vec![TreeItem::new_leaf(Selector::None, get_value_span(root))],
+#[allow(clippy::module_name_repetitions)]
+#[derive(Debug, Clone)]
+pub struct JsonTreeItem<'json> {
+    key: Selector,
+    value: &'json Value,
+    children: Vec<Self>,
+}
+
+impl<'json> JsonTreeItem<'json> {
+    const fn raw_leaf(key: Selector, value: &'json Value) -> Self {
+        Self {
+            key,
+            value,
+            children: Vec::new(),
+        }
+    }
+
+    const fn raw(identifier: Selector, value: &'json Value, children: Vec<Self>) -> Self {
+        // no need for child identifier uniqueness check as JSON itself already prevents this
+        Self {
+            key: identifier,
+            value,
+            children,
+        }
+    }
+
+    #[must_use]
+    pub fn new(root: &'json Value) -> Vec<Self> {
```

Comment:
I think as a lib user I'd expect the normal use case to recurse, and there to be a non recursive alternate. Not 100% sure of what the right naming of the other method would be though (perhaps new_value?)

Reply:
This is the only publish method, the others are private, so their naming is not that important. But yet, a better name for the private ones might be a good idea.

Reply:
Ah - I missed the lack of pub - this is fine then.